l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -286,11 +287,14 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
         require(l2BridgeAddress[_chainId] != address(0), "ShB l2 bridge not deployed");
         bytes32 _assetInfo;
         bytes memory _assetData;
+        bytes32 txDataHash;
+        bool legacyDeposit = false;
```

Comment:
I left a similar question on @kelemeno's PR: what are the security implications of an unintended token being treated as legacy deposit? 

Probably we should not allow assets that are not L1 tokens to use the legacy deposit. 

Also, if the exactly same bridge will be used on all chains (including L2s) to manage transfers to other bridges on another L2s, will it be fine if we use legacy data there also?

Reply:
I would agree and restrict the behaviour strictly to L1 deposits

Reply:
So this is currently only an L1 shared bridge, so we should keep this in mind, but no need to make changes here